discord/automod.py

```diff
@@ -0,0 +1,420 @@
+        exempt_roles: Optional[List[Role]] = None,
+        exempt_channels: Optional[List[Union[GuildChannel, Thread]]] = None,
+    ) -> Self:
+        """|coro|
+
+        This method will edit an existing auto moderation rule.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
+            The new name to change to.
+        event_type: Optional[:class:`AutoModRuleEventType`]
+            The new event type to change to.
+        actions: Optional[List[:class:`AutoModRuleAction`]]
+            The new rule actions to update.
+        enabled: Optional[:class:`bool`]
+            Whether the rule should be enabled or not.
+        exempt_roles: Optional[List[:class:`Role`]]
+            The new roles to exempt from the rule.
+        exempt_channels: Optional[List[Union[:class:`abc.GuildChannel`], :class:`Thread`]]
+            The new channels to exempt from the rule.
+
+        Raises
+        -------
+        :exc:`Forbidden`
+            You are not authorised to perform this action.
+        :exc:`NotFound`
+            The rule was not found.
+
+        Returns
+        --------
+        :class:`AutoModRule`
+            The updated auto moderation rule.
+        """
+        transformed_actions = None
+        if actions:
+            transformed_actions = [action.to_dict() for action in actions]
+
+        data = await self._state.http.modify_auto_moderation_rule(
+            self.guild.id,
+            self.id,
+            name=name,
+            event_type=event_type,
+            actions=transformed_actions,
+            enabled=enabled,
+            exempt_roles=exempt_roles,
+            exempt_channels=exempt_channels,
+        )
+
+        return AutoModRule(data=data, guild=self.guild, state=self._state)
+
+    async def delete(self) -> None:
+        """|coro|
+
+        This method will delete the auto moderation rule.
+
+        Raises
+        -------
+        :exc:`Forbidden`
+            You are not authorised to perform this action.
+        :exc:`NotFound`
+            The rule was not found.
+        """
+        await self._state.http.delete_auto_moderation_rule(self.guild.id, self.id)
+
+
+class AutoModRuleExecution:
+    """Represents an auto moderation rule execution.
+
+    Attributes
+    -----------
+    action: :class:`AutoModRuleAction`
+        The action that was taken.
+    message_id: Optional[:class:`int`]
+        The message ID that was triggered the reaction.
+    rule_id: :class:`int`
+        The ID of the rule that was triggered.
+    rule_trigger_type: :class:`AutoModRuleTriggerType`
+        The trigger type of the rule that was triggered.
+    user_id: :class:`int`
+        The ID of the user that triggered the rule.
+    alert_system_message_id: Optional[:class:`int`]
+        The ID of the system message that was sent to the predefined alert channel.
+    content: :class:`str`
+        The content of the message that triggered the rule.
```

Comment:
Probably mention that this requires the message content intent otherwise it'll always be an empty string.